### PR TITLE
Use urfave/cli package

### PR DESCRIPTION
**Pull request by igungor**
### TODO

- [x] Update README.md
  - [x] Advise installing from prebuilt binaries instead of `go get`. `master` branch is in a constant change right now and we can't guarantee backwards compatibility till v1.0.0 release.
  - [x] Change `Usage` section according to the latest `cli` changes.

### Breaking changes
- `s5cmd get` command is removed. We have 2 identical commands without a clear reasoning. `s5cmd cp <src> .` has the same result as `s5cmd get <src>`.
- `batch-rm` command is removed. Users can run `s5cmd rm` with variadic arguments. Wildcards are accepted (WIP)
- `s5cmd -f` became `s5cmd run`
- All errors are written to stderr
- `s5cmd ls s3://non-existent/object` returns `1` instead of `0`. Related: #23 
- Exit code for errors changed from `127` to `1`
- `s5cmd cp s3://bucket/object local/directory/` now creates the missing `local/directory/` instead of returning an error.

### CODE REVIEW

File: command/cmd_copy.go [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+	dst *objurl.ObjectURL,
+	parents bool,
+) (*objurl.ObjectURL, error) {
+	objname := src.Base()
+	if parents {
+		objname = src.Relative()
+	}
+
+	if originalSrc.HasGlob() {
+		os.MkdirAll(dst.Absolute(), os.ModePerm)
```
Comment (ilkinulas) [inline command/cmd_copy.go]: `os.MkdirAll` returns an error if create dir fails. We should handle it.

**Comment by aykutfarsak** [comment]
Could we add the argument names also? It helps reading code, like 

> What is the meaning of the third param of List() method?

https://github.com/peak/s5cmd/blob/97f7a4940d7c405c6ab1b84aa78a915e59e8dff4/storage/storage.go#L25-L36

**Comment by seruman** [comment]
> Could we add the argument names also? It helps reading code, like
> 
> > What is the meaning of the third param of List() method?
> 
> https://github.com/peak/s5cmd/blob/97f7a4940d7c405c6ab1b84aa78a915e59e8dff4/storage/storage.go#L25-L36

Since implementations omit unused ones, this might be helpful

File: command/cmd_copy.go (hunk 1 above)
Comment (igungor) [reply command/cmd_copy.go]: done.

**Comment by igungor** [comment]
@aykutfarsak @seruman done. please add your comments to that particular line.

File: command/cmd_get.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/peak/s5cmd/storage"
+)
+
+var GetCommand = &cli.Command{
+	Name:     "get",
+	HelpName: "get",
+	Usage:    "TODO",
```
Comment (seruman) [inline command/cmd_get.go]: Missin usage

File: command/cmd_copy.go [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+
+// expandSource returns the full list of objects from the given src argument.
+// If src is an expandable URL, such as directory, prefix or a glob, all
+// objects are returned by walking the source.
+func expandSource(
+	ctx context.Context,
+	src *objurl.ObjectURL,
+	isRecursive bool,
+) (<-chan *storage.Object, error) {
+	// TODO(ig): this function could be in the storage layer.
```
Comment (seruman) [inline command/cmd_copy.go]: +1

Comment (igungor) [reply command/cmd_copy.go]: not going to happen in this PR. looks easy but i think it's not.

File: command/cmd_get.go (hunk 2 above)
Comment (igungor) [reply command/cmd_get.go]: `get` command is gone. Please see the description.

**Pull request merged**